### ISSUE TITLE
Malformed Body Results in Infinite Loop

### ISSUE THREAD

**Issue by jasonmcaffee** [LeisureLink/magicbus#28]
If the body is malformed, MAGIC BUS gets stuck in an infinite loop.

example body:
```
{
  oops: "calling JSON.parse on this object throws an exception"
}
```

results in infinite loop:
```
2016-02-03T17:40:55.701Z - info: debug MAGIC BUS: Beginning consumption from queue imp.payout-service.subscribe
2016-02-03T17:40:55.703Z - info: warn MAGIC BUS: AMQP Connection failed with error.
2016-02-03T17:40:55.703Z - info: info MAGIC BUS: AMQP Connection closed
2016-02-03T17:40:55.703Z - info: info MAGIC BUS: Connecting to docker.dev:5672
2016-02-03T17:40:55.739Z - info: debug MAGIC BUS: Asserting route subscribe
2016-02-03T17:40:55.745Z - info: debug MAGIC BUS: Route subscribe asserted
2016-02-03T17:40:55.745Z - info: debug MAGIC BUS: Beginning consumption from queue imp.payout-service.subscribe
2016-02-03T17:40:55.748Z - info: warn MAGIC BUS: AMQP Connection failed with error.
2016-02-03T17:40:55.749Z - info: info MAGIC BUS: AMQP Connection closed
2016-02-03T17:40:55.749Z - info: info MAGIC BUS: Connecting to docker.dev:5672
2016-02-03T17:40:55.789Z - info: debug MAGIC BUS: Asserting route subscribe
2016-02-03T17:40:55.791Z - info: debug MAGIC BUS: Route subscribe asserted
2016-02-03T17:40:55.791Z - info: debug MAGIC BUS: Beginning consumption from queue imp.payout-service.subscribe
2016-02-03T17:40:55.793Z - info: warn MAGIC BUS: AMQP Connection failed with error.
2016-02-03T17:40:55.794Z - info: info MAGIC BUS: AMQP Connection closed
2016-02-03T17:40:55.794Z - info: info MAGIC BUS: Connecting to docker.dev:5672
2016-02-03T17:40:55.838Z - info: debug MAGIC BUS: Asserting route subscribe
2016-02-03T17:40:55.841Z - info: debug MAGIC BUS: Route subscribe asserted
2016-02-03T17:40:55.841Z - info: debug MAGIC BUS: Beginning consumption from queue imp.payout-service.subscribe
2016-02-03T17:40:55.843Z - info: warn MAGIC BUS: AMQP Connection failed with error.
2016-02-03T17:40:55.844Z - info: info MAGIC BUS: AMQP Connection closed
2016-02-03T17:40:55.844Z - info: info MAGIC BUS: Connecting to docker.dev:5672
2016-02-03T17:40:55.889Z - info: debug MAGIC BUS: Asserting route subscribe
2016-02-03T17:40:55.891Z - info: debug MAGIC BUS: Route subscribe asserted
2016-02-03T17:40:55.891Z - info: debug MAGIC BUS: Beginning consumption from queue imp.payout-service.subscribe
2016-02-03T17:40:55.894Z - info: warn MAGIC BUS: AMQP Connection failed with error.
2016-02-03T17:40:55.894Z - info: info MAGIC BUS: AMQP Connection closed
2016-02-03T17:40:55.894Z - info: info MAGIC BUS: Connecting to docker.dev:5672
```

![image](https://cloud.githubusercontent.com/assets/1034314/12791148/e464413e-ca62-11e5-836f-16260c90e984.png)

**Comment by twindagger** [comment]
Does this still happen?

**Comment by twindagger** [comment]
Fixed in 2.2.0 (i didn't test any earlier versions, but I don't see a reason why this wouldn't be fixed in 2.0)
